backend/authentication/models/user.py

```diff
@@ -192,3 +194,55 @@ def get_associations_with_permission(self, permission: str) -> set:
         from associations.models.role import Role
 
         return set(Role.objects.filter(query).values_list("association__pk", flat=True))
+
+    def get_absolute_url(self):
+        return '/profils/'+self.id
+
+    ## For students' genealogy ##
+
+    def genealogy_relations(self):
+        """All neighbors of a student in the Minor's graph"""
+        successors = []
+        if self.roommate.all:
+            successors.extend(self.roommate.all())
+        if self.minesparent.all:
+            successors.extend(self.minesparent.all())
+        if self.fillots.all:
+            successors.extend(self.fillots.all())
+        if self.cousinast.all:
+            successors.extend(self.cousinast.all())
+        return successors
+
+    def relation_with(self, student):
+        """
+            Describes relation linking self with another student
+            Returns None if the 2 students have no direct link
+        """
+        if self in student.roommate.all():
+            return "co"
+        if self in student.minesparent.all():
+            return "parraine"
+        if self in student.fillots.all():
+            return "fillot.e"
+        if self in student.cousinast.all():
+            return "cousin.e"
+        return None
+
+    @staticmethod
+    def BFS(start, end):
+        """Breadth-First-Search algorithm, to find the shortest path between two students"""
+        visited = []
```

Comment:
using a `set` will provide `O(log n)` complexity vs `O(n)` with a plain array in the `in` in line 242